prunner.go

```diff
@@ -665,31 +676,28 @@ func (r *PipelineRunner) requestPersist() {
 
 func (r *PipelineRunner) CancelJob(id uuid.UUID) error {
 	r.mx.Lock()
+	defer r.mx.Unlock()
 
 	job, ok := r.jobsByID[id]
 	if !ok {
-		r.mx.Unlock()
 		return ErrJobNotFound
 	}
 
 	if job.Completed {
-		r.mx.Unlock()
 		return errJobAlreadyCompleted
 	}
 
 	if job.Start == nil {
-		r.mx.Unlock()
 		return errJobNotStarted
 	}
 
-	// Unlock mutex before calling cancel to prevent deadlocks from state updates
-	r.mx.Unlock()
+	log.
+		WithField("component", "runner").
+		WithField("pipeline", job.Pipeline).
+		WithField("jobID", job.ID).
+		Debugf("Canceling job")
 
-	// SAFEGUARD: it could happen that a job is cancelled which has never been scheduled.
-	// thus, we need to check for the existence of the job scheduler before cancelling.
-	if job.sched != nil {
-		job.sched.Cancel()
-	}
+	go job.sched.Cancel()
```

Comment:
This should do the trick of making sure `job.sched` is not nil and keep the go routine from deadlocking on status updates while calling `Cancel()`.